### ISSUE TITLE
DistMult: support normalising embeddings

### ISSUE THREAD

**Issue by huonw** [stellargraph/stellargraph#980]
### Description

The DistMult paper (https://arxiv.org/abs/1412.6575) states that they normalise the computed embeddings after each epoch, and, in particular, that it helps.

> The entity vectors are renormalized to have unit length after each gradient step (it is an effective technique that empirically improved all the models).

Unfortunately, we cannot do this easily at the moment, because TF Keras's `Embeddings(..., embeddings_constraint=...)` doesn't actually seem to work: tensorflow/tensorflow#33755 .

See also: https://github.com/stellargraph/stellargraph/pull/865#discussion_r384172379

### Done Checklist

- [ ] Produced code for required functionality
- [ ] Tests written and coverage checked
- [ ] Code review performed
- [ ] Documentation on Google Docs  (if applicable)
- [ ] Documentation in repo
- [ ] Version number reflects new status
- [ ] CHANGELOG.md updated
- [ ] Team demo
